### ISSUE TITLE
wrong cors error with AWS::Serverless::API

### ISSUE THREAD

**Issue by gojko** [aws-cloudformation/cfn-python-lint#698]
*cfn-lint version*: 0.15.0

*Description of issue*: CFN Lint complains about a perfectly valid SAM stack (deploys successfully using CloudFormation), when Cors configuration is turned on. 

The error is: `E0001 Resource with id [WebApi] is invalid. Cors works only with inline Swagger specified in 'DefinitionBody' property`

a minimal template displaying the error:

```
AWSTemplateFormatVersion: '2010-09-09'
Transform: AWS::Serverless-2016-10-31
Parameters:
  AppStage:
    Type: String
Resources:
  WebAssetsS3Bucket:
    Type: AWS::S3::Bucket
    Properties:
      AccessControl: PublicRead
      WebsiteConfiguration:
        ErrorDocument: 404.html
        IndexDocument: index.html
  WebApi:
    Type: AWS::Serverless::Api
    Properties:
      StageName: !Ref AppStage
      Cors: !Sub "'${WebAssetsS3Bucket.WebsiteURL}'"
```

**Comment by kddejong** [comment]
We use SAM translator to translate and process the completed template.   We are getting that error from SAM but I'm not sure why yet.

https://github.com/awslabs/serverless-application-model/blob/cbd4d9ad40a71b838f1e72b3b960689f30890bf9/samtranslator/model/api/api_generator.py#L188-L191

**Comment by kddejong** [comment]
@gojko does this sum up your issue? https://github.com/awslabs/aws-sam-cli/issues/803

**Comment by gojko** [comment]
@kddejong I guess so, looks like the same error. The template I used does not provide an API definition, so I assume the implicit one will be used.

**Comment by kddejong** [comment]
I think we are going to have to plus 1 that issue as it also appears that `sam validate` doesn't work on this template.
```
➜  cfn-python-lint git:(master) ✗ sam validate --template sam.yaml                                                                                                        
Template provided at './sam1.yaml' was invalid SAM Template.
Error: [InvalidResourceException('WebApi', "Cors works only with inline Swagger specified in 'DefinitionBody' property")] ('WebApi', "Cors works only with inline Swagger specified in 'DefinitionBody' property")
```

**Comment by gojko** [comment]
ok, i'll close this one
